### PR TITLE
Enable linting for __tests__

### LINKED ISSUE #3971: Lint tests
After #3968, the next thing we should do is start linting our tests. Historically we've ignored them due to lack of parser compatibility. But that shouldn't be a problem anymore. We may want to integrate https://www.npmjs.com/package/eslint-plugin-react to more aggressively lint our JSX in tests.

cc @PiPeep

### PR DESCRIPTION
**Pull request by PiPeep**
Closes #3971.

> After #3968, the next thing we should do is start linting our tests.
> Historically we've ignored them due to lack of parser compatibility.
> But that shouldn't be a problem anymore. We may want to integrate
> https://www.npmjs.com/package/eslint-plugin-react to more aggressively
> lint our JSX in tests.

I understand this diff touches a lot of stuff, so I tried to keep it to
a near-minimal set of changes to make eslint happy. If you're wondering
why I changed something, let me know and I'll try to map it up to the
rule that I was fixing it for.

### CODE REVIEW

**Comment by PiPeep** [comment]
Potential future steps:

- Enable more of the eslint-plugin-react rules.
- Make our grunt task smarter to be able to pattern-match `__tests__`, and provide alternate .eslintrc files for tests and normal code. E.g. I'd propose disabling `no-unused-expressions` for tests.
- Fix all of the 80-column warnings (ugh)

File: npm-react/addons.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 var warning = require('./lib/warning');
 warning(
   false,
-  'require("react/addons") is deprecated. ' +
-  'Access using require("react/addons/{addon}") instead.'
+  'require(\'react/addons\') is deprecated. ' +
```
Comment (spicyj) [inline npm-react/addons.js]: Why is this necessary?

File: src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js [hunk 2]
```diff
@@ -50,20 +50,20 @@ describe('ReactComponentWithPureRenderMixin', function() {
       getInitialState: function() {
         return {
           cut: false,
-          slices: 1,
+          slices: 1
```
Comment (spicyj) [inline src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js]: I'd rather just turn off the no trailing commas rule.

File: src/addons/__tests__/update-test.js [hunk 3]
```diff
@@ -72,7 +72,10 @@ describe('update', function() {
   });
 
   it('should support apply', function() {
-    expect(update(2, {$apply: function(x) { return x * 2; }})).toEqual(4);
+    function f(x) {
```
Comment (spicyj) [inline src/addons/__tests__/update-test.js]: What was the warning here

File: src/addons/transitions/__tests__/ReactTransitionChildMapping-test.js [hunk 4]
```diff
@@ -109,6 +109,7 @@ describe('ReactTransitionChildMapping', function() {
     });
   });
 
+  /*eslint-disable no-undef-init */
```
Comment (spicyj) [inline src/addons/transitions/__tests__/ReactTransitionChildMapping-test.js]: We should either change this to `var next;` below or just turn off this rule everywhere if we don't care.

File: src/addons/transitions/__tests__/ReactTransitionGroup-test.js [hunk 5]
```diff
@@ -128,7 +126,7 @@ describe('ReactTransitionGroup', function() {
       },
       render: function() {
         var children = [];
-        for (var i = 0; i < this.state.count; i++) {
+        for (let i = 0; i < this.state.count; i++) {
```
Comment (spicyj) [inline src/addons/transitions/__tests__/ReactTransitionGroup-test.js]: We don't support `let` internally at FB yet so this can't go in like this.

File: npm-react/addons.js (hunk 1 above)
Comment (PiPeep) [inline npm-react/addons.js]: That isn't actually. My bad regexp codemod caught it, and I figured it wouldn't hurt to change it. Let me know if you want me to revert it.

File: src/addons/transitions/__tests__/ReactTransitionGroup-test.js [hunk 6]
```diff
@@ -94,15 +92,15 @@ describe('ReactTransitionGroup', function() {
 
   it('should handle enter/leave/enter/leave correctly', function() {
     var log = [];
-    var cb;
+    var willEnterCb;
```
Comment (spicyj) [inline src/addons/transitions/__tests__/ReactTransitionGroup-test.js]: (Why this change?)

File: src/isomorphic/children/__tests__/ReactChildren-test.js [hunk 7]
```diff
@@ -431,13 +429,14 @@ describe('ReactChildren', function() {
     expect(console.error.calls[0].args[0]).toContain('use of a keyed object');
   });
 
+  /*eslint-disable no-unused-expressions */
```
Comment (spicyj) [inline src/isomorphic/children/__tests__/ReactChildren-test.js]: Instead of disabling this rule, maybe we chould use the `void` operator on each place that triggers it (which this rule's docs claim doesn't warn). Like `void frag[key];`.

File: src/isomorphic/children/__tests__/ReactChildren-test.js [hunk 8]
```diff
@@ -227,7 +223,7 @@ describe('ReactChildren', function() {
 
     var instance = (
       <div>{
-        [frag({
+        [ReactFragment.create({
```
Comment (spicyj) [inline src/isomorphic/children/__tests__/ReactChildren-test.js]: (Why?)

File: src/isomorphic/classic/class/__tests__/ReactClass-test.js [hunk 9]
```diff
@@ -189,9 +190,8 @@ describe('ReactClass-spec', function() {
       'mean shouldComponentUpdate()? The name is phrased as a question ' +
       'because the function is expected to return a value.'
     );
-
-    <NamedComponent />; // Shut up lint
```
Comment (spicyj) [inline src/isomorphic/classic/class/__tests__/ReactClass-test.js]: For this one, we could just do `displayName: 'NamedComponent',` inside the class spec and drop the var assignment.

File: src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js (hunk 2 above)
Comment (PiPeep) [inline src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js]: I agree, and think we should actually enforce trailing commas, but I think that should be another commit, and could be done with a codemod.

File: src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js [hunk 10]
```diff
@@ -263,7 +261,7 @@ describe('ReactClass-mixin', function() {
     expect(Component.abc).toBe('def');
   });
 
-  it("should throw if mixins override each others' statics", function() {
+  it('should throw if mixins override each others\' statics', function() {
```
Comment (spicyj) [inline src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js]: Let's do avoid-escape http://eslint.org/docs/rules/quotes.html and then leave things like these as double quotes.

File: src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js [hunk 11]
```diff
@@ -368,24 +372,24 @@ describe('ReactClass-mixin', function() {
   });
 
   it('should include the mixin keys in even if their values are falsy',
-    function() {
-      var mixin = {
-        keyWithNullValue: null,
-        randomCounter: 0
-      };
+      function() {
```
Comment (spicyj) [inline src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js]: Neat, I'm surprised this warns.

File: src/isomorphic/classic/element/__tests__/ReactElementValidator-test.js [hunk 12]
```diff
@@ -283,7 +286,7 @@ describe('ReactElementValidator', function() {
   });
 
   it('gives a helpful error when passing null, undefined, boolean, or number',
-      () => {
```
Comment (spicyj) [inline src/isomorphic/classic/element/__tests__/ReactElementValidator-test.js]: Did this warn? Why?

File: src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js (hunk 11 above)
Comment (spicyj) [inline src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js]: Maybe we should just keep it on one line and ignore the line length.

File: src/addons/transitions/__tests__/ReactTransitionChildMapping-test.js (hunk 4 above)
Comment (PiPeep) [inline src/addons/transitions/__tests__/ReactTransitionChildMapping-test.js]: I think we should turn it off, but only for tests. Explicitly writing `var next = undefined` makes the code more readable. That requires making our grunt rule a bit smarter so it can pattern match `__tests__` directories, which should be another commit and codemod.

Let me know if you'd prefer I turn it off globally (for normal code and tests) and I'll do that instead.

File: src/addons/transitions/__tests__/ReactTransitionGroup-test.js (hunk 6 above)
Comment (PiPeep) [inline src/addons/transitions/__tests__/ReactTransitionGroup-test.js]: `componentWillLeave` also defined `cb` so there was shadowing. Additionally, the existence of `_cb` and `cb` seemed a bit ugly. This seemed like the smallest modification I could make that didn't sacrifice readability.

File: src/renderers/dom/client/__tests__/ReactMount-test.js [hunk 13]
```diff
@@ -169,25 +169,27 @@ describe('ReactMount', function() {
     );
   });
 
-  (WebComponents === undefined ? xit : it)
```
Comment (spicyj) [inline src/renderers/dom/client/__tests__/ReactMount-test.js]: What was wrong with the old code here?

File: src/isomorphic/children/__tests__/ReactChildren-test.js (hunk 7 above)
Comment (PiPeep) [inline src/isomorphic/children/__tests__/ReactChildren-test.js]: I like this idea. I'll grep for `/*eslint-disable no-unused-expressions */` and do that.

File: src/renderers/shared/event/eventPlugins/__tests__/ResponderEventPlugin-test.js [hunk 14]
```diff
@@ -218,23 +215,27 @@ var registerTestHandlers = function(eventTestConfig, readableIDToID) {
...
     );
   };
+  /*eslint-disable no-loop-func */
   var registerOneEventType = function(registrationName, eventTypeTestConfig) {
-    for (var readableID in eventTypeTestConfig) {
-      var nodeConfig = eventTypeTestConfig[readableID];
-      var id = readableIDToID[readableID];
-      var handler = nodeConfig.order === NA ? neverFire.bind(null, readableID, registrationName) :
-        function(readableID, registrationName, nodeConfig, e) {
+    for (const readableID in eventTypeTestConfig) {
```
Comment (spicyj) [inline src/renderers/shared/event/eventPlugins/__tests__/ResponderEventPlugin-test.js]: (Can't use const.)

File: src/isomorphic/children/__tests__/ReactChildren-test.js (hunk 8 above)
Comment (PiPeep) [inline src/isomorphic/children/__tests__/ReactChildren-test.js]: Later in the code, there's the test:

```js
   it('should warn if a fragment is accessed', function() {
     spyOn(console, 'error');
     var child = React.createElement('span');
     var frag = ReactChildren.map([child, child], function(c) {
       return c;
     });
     ...
```

Which means there's shadowing. Since frag was just a stub method used in only two places, I figured it'd be better to just get rid of it rather than renaming things to avoid a conflict.

File: src/renderers/shared/reconciler/__tests__/ReactCompositeComponentError-test.js [hunk 15]
```diff
@@ -26,7 +25,9 @@ describe('ReactCompositeComponent-error', function() {
       }
     });
 
+    /*eslint-disable no-new*/
```
Comment (spicyj) [inline src/renderers/shared/reconciler/__tests__/ReactCompositeComponentError-test.js]: Let's just turn this off globally.

File: src/renderers/shared/reconciler/__tests__/ReactCompositeComponentNestedState-test.js [hunk 16]
```diff
@@ -88,10 +88,12 @@ describe('ReactCompositeComponentNestedState-state', function() {
 
     var logger = mocks.getMockFunction();
 
-    var instance = React.render(
+    /*eslint-disable no-unused-expressions */
```
Comment (spicyj) [inline src/renderers/shared/reconciler/__tests__/ReactCompositeComponentNestedState-test.js]: I don't think you need this?

File: src/renderers/shared/reconciler/__tests__/ReactIdentity-test.js [hunk 17]
```diff
@@ -107,12 +107,12 @@ describe('ReactIdentity', function() {
     var Wrapper = React.createClass({
 
       render: function() {
-        var span1 = <span ref="span1" key={key} />;
-        var span2 = <span ref="span2" />;
+        var s1 = <span ref="span1" key={key} />;
```
Comment (spicyj) [inline src/renderers/shared/reconciler/__tests__/ReactIdentity-test.js]: Why the change?

File: src/renderers/shared/reconciler/__tests__/ReactUpdates-test.js [hunk 18]
```diff
@@ -785,7 +787,6 @@ describe('ReactUpdates', function() {
         return <div />;
       },
       componentDidUpdate: function() {
-        var component = this;
```
Comment (spicyj) [inline src/renderers/shared/reconciler/__tests__/ReactUpdates-test.js]: Did you mean to delete this?

File: src/addons/__tests__/update-test.js (hunk 3 above)
Comment (PiPeep) [inline src/addons/__tests__/update-test.js]: eslint requires indenting code blocks, instead of doing them on the same line. This is part of [`brace-style`](http://eslint.org/docs/rules/brace-style). If you'd prefer to allow single-line function declarations, we can enable `allowSingleLine`.

File: src/addons/transitions/__tests__/ReactTransitionGroup-test.js (hunk 5 above)
Comment (PiPeep) [inline src/addons/transitions/__tests__/ReactTransitionGroup-test.js]: It's been brought up multiple times that we shouldn't be running these tests internally anyways, so I'll look into that issue before we merge this.

**Comment by spicyj** [comment]
This looks pretty good overall.

File: npm-react/addons.js (hunk 1 above)
Comment (spicyj) [inline npm-react/addons.js]: I guess let's keep it as single quotes in the require but use double quotes for the outer string so we don't need the escapes.

File: src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js (hunk 2 above)
Comment (spicyj) [inline src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js]: In this commit can we turn off the rule and not remove all the trailing commas (just to add them again a little later)? Also happy for us to add them everywhere before landing this PR if you'd rather do that.

File: src/addons/__tests__/update-test.js (hunk 3 above)
Comment (spicyj) [inline src/addons/__tests__/update-test.js]: Does `(x) => x * 2` work here? If so, let's do that – but otherwise this is okay.

File: src/addons/transitions/__tests__/ReactTransitionChildMapping-test.js (hunk 4 above)
Comment (spicyj) [inline src/addons/transitions/__tests__/ReactTransitionChildMapping-test.js]: Let's just turn it off globally.

File: src/renderers/shared/reconciler/__tests__/ReactUpdates-test.js (hunk 18 above)
Comment (PiPeep) [inline src/renderers/shared/reconciler/__tests__/ReactUpdates-test.js]: Yes. `component` was being shadowed (see line 802), but I both declarations held the same value. This part of the test is checking `asap` sets `this` correctly, which I believe this version still checks for.

File: src/addons/transitions/__tests__/ReactTransitionGroup-test.js (hunk 6 above)
Comment (spicyj) [inline src/addons/transitions/__tests__/ReactTransitionGroup-test.js]: :+1:

File: src/isomorphic/children/__tests__/ReactChildren-test.js (hunk 8 above)
Comment (spicyj) [inline src/isomorphic/children/__tests__/ReactChildren-test.js]: Sure.

File: src/renderers/shared/reconciler/__tests__/ReactUpdates-test.js (hunk 18 above)
Comment (spicyj) [inline src/renderers/shared/reconciler/__tests__/ReactUpdates-test.js]: Ah, I missed the declaration below. :+1:

File: src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js (hunk 11 above)
Comment (PiPeep) [inline src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js]: There's other tests where we're breaking lines like this, so I think it's fine.

Is trying to hit 80 columns everywhere a goal or not? Currently it's making the linter incredibly noisy. (235 warnings)

File: src/isomorphic/classic/element/__tests__/ReactElementValidator-test.js (hunk 12 above)
Comment (PiPeep) [inline src/isomorphic/classic/element/__tests__/ReactElementValidator-test.js]: This is a workaround for eslint/eslint#2239. It had @zpao and I scratching our heads for a bit.

File: src/renderers/shared/reconciler/__tests__/ReactIdentity-test.js (hunk 17 above)
Comment (PiPeep) [inline src/renderers/shared/reconciler/__tests__/ReactIdentity-test.js]: `no-shadow`. On line 121 and 122:

```js
var span1 = instance.refs.span1;
var span2 = instance.refs.span2;
```

As a general rule, I prefer giving the shorter name to the more deeply-nested bit, so these instances got changed.

File: src/renderers/dom/client/__tests__/ReactMount-test.js (hunk 13 above)
Comment (PiPeep) [inline src/renderers/dom/client/__tests__/ReactMount-test.js]: There's whitespace between the function identifier and the open-paren designating the argument list.

File: src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js (hunk 11 above)
Comment (spicyj) [inline src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js]: I dunno. I think it would be best if you'd get a lint warning on new lines that are too long but that we could still commit long lines if we wanted to. That's hard to make work with our current setup though.

I like the line length limit for reading actual code but it doesn't feel useful to me on lines like this.

Comment (PiPeep) [inline src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js]: @zpao, thoughts?

Comment (PiPeep) [inline src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js]: What if we had a flag you could pass to `grunt lint` to hide warnings? It wouldn't solve the problem of discouraging people from using long lines of code, but it would at least make it easier to sort through the errors.

File: src/renderers/dom/client/__tests__/ReactMount-test.js (hunk 13 above)
Comment (spicyj) [inline src/renderers/dom/client/__tests__/ReactMount-test.js]: Hm, okay.

File: src/renderers/shared/reconciler/__tests__/ReactIdentity-test.js (hunk 17 above)
Comment (spicyj) [inline src/renderers/shared/reconciler/__tests__/ReactIdentity-test.js]: Ahh, I keep looking above but not below. This is fine then; you could also just inline them.

File: src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js (hunk 2 above)
Comment (PiPeep) [inline src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js]: I'll open another PR reversing this rule, and rebase this PR on top of it after it merges. That seems like the least amount of overall effort.

File: src/addons/__tests__/update-test.js (hunk 3 above)
Comment (PiPeep) [inline src/addons/__tests__/update-test.js]: :+1: That worked. Strange, I could've sworn I tried that the first time around and it caused some other error. Maybe I did something else wrong.

File: src/renderers/shared/reconciler/__tests__/ReactCompositeComponentNestedState-test.js (hunk 16 above)
Comment (PiPeep) [inline src/renderers/shared/reconciler/__tests__/ReactCompositeComponentNestedState-test.js]: It was triggering an `unused-var` warning. Regardless, the `void` change fixes it.

File: src/renderers/shared/reconciler/__tests__/ReactIdentity-test.js (hunk 17 above)
Comment (PiPeep) [inline src/renderers/shared/reconciler/__tests__/ReactIdentity-test.js]: Yeah, I think most of the `no-shadow` rules triggered in places where the variable was defined below the shadowing scope because they probably would've been caught by a human earlier otherwise.

File: src/renderers/shared/reconciler/__tests__/ReactCompositeComponentError-test.js (hunk 15 above)
Comment (PiPeep) [inline src/renderers/shared/reconciler/__tests__/ReactCompositeComponentError-test.js]: Changing it to use `void` like the `no-unused-expressions` cases seems to work for this. Would that be preferable? It's hard for me to think of any good places (besides tests) that you'd want to use constructor side-effects.

File: .eslintrc [hunk 19]
```diff
@@ -1,12 +1,27 @@
...
+  - react
+
 env:
   browser: true
   node: true
 
 globals:
   __DEV__: true
+  # Jest / Jasmine
+  describe: false
```
Comment (zpao) [inline .eslintrc]: Any difference between true and false here? If not let's be consistent (I don't care what the value is if it works)

Comment (PiPeep) [inline .eslintrc]: > To configure global variables inside of a configuration file, use the globals key and indicate the global variables you want to use. Set each global variable name equal to `true` to allow the variable to be overwritten or `false` to disallow overwriting.

http://eslint.org/docs/user-guide/configuring.html

File: src/addons/__tests__/ReactFragment-test.js [hunk 20]
```diff
@@ -27,7 +27,7 @@ describe('ReactFragment', function() {
       x: <span />,
       y: <span />
     };
-    <div>{children}</div>;
+    void <div>{children}</div>;
```
Comment (zpao) [inline src/addons/__tests__/ReactFragment-test.js]: Eh, not sure how I feel about this. I don't have a good alternative though

File: src/addons/transitions/__tests__/ReactTransitionGroup-test.js (hunk 5 above)
Comment (zpao) [inline src/addons/transitions/__tests__/ReactTransitionGroup-test.js]: We already don't run the tests internally. But we still lint internally and I wouldn't be surprised if we end up choking on let. Regardless though, let's not introduce the first `let` here.

File: src/addons/__tests__/ReactFragment-test.js (hunk 20 above)
Comment (PiPeep) [inline src/addons/__tests__/ReactFragment-test.js]: The alternative, IMHO is to disable no-unused-expression in `__tests__` only, by extending the grunt task to do the pattern matching itself, or by supporting multiple sets of eslintrc and eslintignore files.

File: src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js (hunk 11 above)
Comment (zpao) [inline src/isomorphic/classic/class/__tests__/ReactClassMixin-test.js]: I'm leaning more and more towards a custom line-length rule (ignore length on `it` lines and docblocks). The ideal case is what Ben said where we only show the output for "changed" sections of code, perhaps we'll get there.

For now I don't think we should go out of our way to fix line length but fixing some along the way is fine.

File: src/isomorphic/classic/types/__tests__/ReactPropTypes-test.js [hunk 21]
```diff
@@ -789,8 +789,8 @@ describe('ReactPropTypes', function() {
       var instance = <Component num={6} />;
       instance = ReactTestUtils.renderIntoDocument(instance);
       expect(console.error.argsForCall.length).toBe(1);
-      expect(console.error.argsForCall[0][0]).toBe(
-      'Warning: Failed propType: num must be 5!');
+      expect(console.error.argsForCall[0][0])
+        .toBe('Warning: Failed propType: num must be 5!');
```
Comment (zpao) [inline src/isomorphic/classic/types/__tests__/ReactPropTypes-test.js]: I think our style leans a bit more towards
```js
    expect(console.error.argsForCall[0][0]).toBe(
      'Warning: Failed propType: num must be 5!'
    );
```

File: vendor/constants.js [hunk 22]
```diff
@@ -44,7 +44,7 @@ module.exports = function(babel) {
           // into this:
           //
           // if (!condition) {
-          //   if ("production" !== process.env.NODE_ENV) {
+          //   if ('production' !== process.env.NODE_ENV) {
```
Comment (zpao) [inline vendor/constants.js]: Can you revert this file (it actually does end up as double quotes because babel)

Comment (PiPeep) [inline vendor/constants.js]: Good catch.

File: src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js (hunk 2 above)
Comment (PiPeep) [inline src/addons/__tests__/ReactComponentWithPureRenderMixin-test.js]: Okay. I've changed my mind here. I'll fix the dangling commas after this PR lands. I hacked together a fancy codemod tool that hooks into eslint's node API to fix the code. (which is awesome because we get zero false positives, and it obeys eslintignore/eslintrc files)

**Comment by PiPeep** [comment]
@zpao @spicyj

I think I fixed everything except https://github.com/facebook/react/pull/3985/files#r31360729 , which I'd like an opinion on, and the trailing comma issue, which I'll fix in a later PR, after this one is merged.

I could revert the comma changes I made here and disable the rule right now, but there's not really any point besides slightly shortening the diff.

File: .eslintrc (hunk 19 above)
Comment (zpao) [inline .eslintrc]: Ah no, I just didn't realize there was a difference. I might have known back when I first did this (I think there's a test that toggles `__DEV__`)

File: src/renderers/shared/reconciler/__tests__/ReactCompositeComponentError-test.js (hunk 15 above)
Comment (zpao) [inline src/renderers/shared/reconciler/__tests__/ReactCompositeComponentError-test.js]: If we find ourselves needing the rule disabled for other cases let's do that but the single case disable is fine. I don't care how we do it - either is fine by me.

File: src/isomorphic/classic/class/__tests__/ReactClass-test.js [hunk 23]
```diff
@@ -159,6 +159,7 @@ describe('ReactClass-spec', function() {
     }
   });
 
+  /*eslint-disable no-unused-vars */
```
Comment (spicyj) [inline src/isomorphic/classic/class/__tests__/ReactClass-test.js]: I don't think you need this now?

File: src/isomorphic/classic/element/__tests__/ReactElement-test.js [hunk 24]
```diff
@@ -285,7 +283,7 @@ describe('ReactElement', function() {
 
   it('warns when adding a prop after element creation', function() {
     spyOn(console, 'error');
-    var el = document.createElement('div');
+    var elContainer = document.createElement('div');
```
Comment (spicyj) [inline src/isomorphic/classic/element/__tests__/ReactElement-test.js]: We use `container` in a bunch of other places which sounds nicer here, I think.

**Comment by spicyj** [comment]
lgtm – can you fix those two things? Then feel free to squash and merge.
